dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -941,6 +944,17 @@ public SourceFile getSourceFile(String path, String id, String branch, Descripto
         throw new UnsupportedOperationException("not implemented/needed for github");
     }
 
+    @Override
+    public List<SourceControlOrganization> getOrganizations() {
+        try {
+            return github.getMyOrganizations().entrySet().stream()
+                    .map(o -> new SourceControlOrganization(o.getValue().getId(), o.getKey())).collect(Collectors.toList());
+        } catch (IOException e) {
+            LOG.info(githubTokenUsername + ": Cannot retrieve their organizations", e);
```

Comment:
> Actually, think we get this for free already since...

That's true, but here it's swallowing IOException (not sure when they will occur), and if there is not a GitHub token associated with the account, it returns an empty array list on line 955. Both are sort of edge cases, since almost all our users have GitHub accounts.

Reply:
I don't think they can get here. A bad token should not result in a IOException, the check occurs under https://github.com/dockstore/dockstore/pull/4171/files#diff-2c8f98ad7fbc26776c9437aeb9dfe12d07fe60de2ada259bf78564e92198295aR811 which is a CustomWebApplicationException which isn't related to IOException. That's why the test is getting a 400 and not a 500? (not sure what a IOException works out to)

Reply:
Oh, I see, you mentioned two cases in the original comment, bad token and no token. 

No token is a bit funny. A user that intentionally un-linked their github account or logged in through Google. Hmm